java/ql/src/experimental/Security/CWE/CWE-094/GroovyInjectionBlacklist.java

```diff
@@ -0,0 +1,90 @@
+import groovy.lang.GroovyClassLoader;
+import groovy.lang.GroovyShell;
+import groovy.lang.Script;
+
+/**
+ * IMPORTANT: 
+ * Besides your created blacklist also add these classes there:
+ * 
+ * groovy.transform.ASTTest
+ * GrabConfig
+ * org.buildobjects.process.ProcBuilder
+ * 
+ * for compile-time RCE protection
+ */
+public class SandboxGroovyClassLoader extends ClassLoader {
+
+    public SandboxGroovyClassLoader(ClassLoader parent) {
+        super(parent);
+    }
+
+    @Override
+    public Class<?> loadClass(String name) throws ClassNotFoundException {
+        // Class names BLACKLIST
+        if (name.startsWith("java.lang.System"))
+            return null;
+        return super.loadClass(name);
+    }
+
+    @Override
+    protected Class<?> loadClass(String name, boolean resolve) throws ClassNotFoundException {
+        // Class names BLACKLIST
+        if (name.startsWith("java.lang.System"))
+            return null;
+        return super.loadClass(name, resolve);
+    }
+
+    static void runWithGroovyClassLoader() throws Exception {
+        System.out.println("Begin runWithGroovyClassLoader");
+
+        String code = "def hello_world() { java.lang.System.currentTimeMillis(); };";
+
+        GroovyClassLoader groovyClassLoader = new GroovyClassLoader();
+        Class<?> scriptClass = groovyClassLoader.parseClass(code);
+        Object scriptInstance = scriptClass.newInstance();
+        Object result = scriptClass.getDeclaredMethod("hello_world", new Class[]{}).invoke(scriptInstance, new Object[]{});
```

Comment:
Is this example actually safe? It reads like the one below without use of sandboxing? If so, omit it here and show only the good case (the file `GroovyInjectionBad.java` already shows the bad case)